FC-1145: :memo:  Foundations Chapter (#8)
Templum/developers-community: pages/Documents/DeveloperTools/LivePersonFunctions/Foundations/limitations.md

  - function-as-a-service-foundations-limitations.html
---

> This page will offer you with a high level overview about all limitations that are in place. This should help you determine if your use-case is feasible and you want to move forward with the LivePerson Functions platform.

Reviewer: ```suggestion
This page will offer you a high level overview of all limitations that are in place. This should help you determine if your use-case is feasible and you want to move forward with the LivePerson Functions platform.
```

### Whitelisting

* You may not have more than one wildcard
  * `*.*.liveperson.com` is not allowed while `*.liveperson.com` is
* Calls to none allowed domains will be aborted
* Changes to the allowed domain list can take up to 5 minutes before being reflected

> * Domain name needs to be valid based on [RFC1035](https://datatracker.ietf.org/doc/html/rfc1035)

Reviewer: ```suggestion
* Domain names need to be valid based on [RFC1035](https://datatracker.ietf.org/doc/html/rfc1035)
```

### Function

* Source Code must not exceed 100.000 characters
* Multiple requests are handled by a single functions instance
* Resources of a function instance are limited to 500m units of CPU and 256 MB of RAM
  * Exceeding the resource limit will result in the immediate stopping of the function, failing any request being processed by it
* Only the last 5 deployments will be stored
* Execution time is limited to 30s, some [event sources](liveperson-functions-event-sources-overview.html) may have a lower time limit

#### Timeouts

> This is not a direct limitation, but a strong discouragement from our side. Please avoid the usage of `setTimeout` in your code. Given that a single function instance me handle more than one request, having timeouts running may negatively impact performance. Especially when the function is experiencing high traffic volumes. Generally any coding that blocks or fills up the event loop is something that should be avoided. If that is a deal breaker for you, we advice to look into an alternative solution.

Reviewer: ```suggestion
This is not a direct limitation, but a strong discouragement from our side. Please avoid the usage of `setTimeout` in your code. Given that a single function instance may handle more than one request, having timeouts running may negatively impact performance. Especially when the function is experiencing high traffic volumes. Generally, any coding that blocks or fills up the event loop is something that should be avoided. If that is a deal breaker for you, we advise you to look into an alternative solution.
```
